packages/orchestration/src/orchestration-api.ts

```diff
@@ -89,7 +92,11 @@ export interface Chain<CI extends ChainInfo> {
   makeAccount: () => Promise<OrchestrationAccount<CI>>;
   // FUTURE supply optional port object; also fetch port object
 
-  query: CI extends { icqEnabled: true } ? ICQQueryFunction : never;
+  query: CI extends { icqEnabled: true }
+    ? ICQQueryFunction
+    : CI['chainId'] extends `agoric${string}`
+      ? QueryManyFn
```

Comment:
what's the interface that `ICQQueryFunction` and `QueryManyFn` both satisfy? please add a types test to ensure they always both satisfy it

Reply:
There isn't a common interface between the two. I've added a type test demonstrating the different behavior here: https://github.com/Agoric/agoric-sdk/pull/9935/commits/af26af0dfce5f2649ddfa3f2016806204824262f